evm_arithmetization/src/cpu/decode.rs

```diff
@@ -304,14 +311,25 @@ pub(crate) fn eval_ext_circuit<F: RichField + Extendable<D>, const D: usize>(
         yield_constr.constraint(builder, constr);
     }
 
+    let one_extension = builder.constant_extension(F::Extension::ONE);
+    let is_not_kernel_mode = builder.sub_extension(one_extension, kernel_mode);
+
+    // Manually check that the fp254 opertions are kernel-only instructions.
+    let constr = builder.mul_extension(is_not_kernel_mode, lv.op.fp254_op);
+    yield_constr.constraint(builder, constr);
+
+    // Manually check that SUBMOD is a kernel-only instruction. SUBMOD is
+    // differentiated by its second bit set to 1.
+    let submod_op = builder.mul_extension(lv.op.ternary_op, lv.opcode_bits[1]);
+    let constr = builder.mul_extension(submod_op, is_not_kernel_mode);
+    yield_constr.constraint(builder, constr);
```

Comment:
Just realizing it now (applies to the other), but if we were to define `is_not_kernel_mode` as `kernel_mode - P::ONES`, we should be able to use `mul_sub_extension` in the circuit instead?